src/core/meshcop/joiner.cpp

```diff
@@ -95,92 +103,125 @@ otError Joiner::Start(const char *     aPSKd,
     ThreadNetif &   netif = GetNetif();
     otError         error;
     Mac::ExtAddress joinerId;
-    Crc16           ccitt(Crc16::kCcitt);
-    Crc16           ansi(Crc16::kAnsi);
 
-    VerifyOrExit(mState == OT_JOINER_STATE_IDLE, error = OT_ERROR_BUSY);
+    otLogInfoMeshCoP("Joiner starting");
 
-    GetNotifier().Signal(OT_CHANGED_JOINER_STATE);
+    VerifyOrExit(mState == OT_JOINER_STATE_IDLE, error = OT_ERROR_BUSY);
 
-    // use extended address based on factory-assigned IEEE EUI-64
+    // Use extended address based on factory-assigned IEEE EUI-64
     GetJoinerId(joinerId);
     netif.GetMac().SetExtAddress(joinerId);
     netif.GetMle().UpdateLinkLocalAddress();
 
-    for (size_t i = 0; i < sizeof(joinerId); i++)
+    SuccessOrExit(error = netif.GetCoapSecure().Start(kJoinerUdpPort));
+    SuccessOrExit(error = netif.GetCoapSecure().SetPsk(reinterpret_cast<const uint8_t *>(aPSKd),
+                                                       static_cast<uint8_t>(strlen(aPSKd))));
+    SuccessOrExit(error = netif.GetCoapSecure().GetDtls().mProvisioningUrl.SetProvisioningUrl(aProvisioningUrl));
```

Comment:
I think we should submit another PR to move `SetProvisioningUrl()` out of `Dtls`.